# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/ClientWebGLContext.cpp
# Commit: 71ad2ed8e9ba
# Full Hash: 71ad2ed8e9ba668e11656652ebd4235ef64565c2
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-11 04:04:33
# Regressor Bug: 1632249
# File Overlap Count: 1
# Description:
#   Bug 1632249 - Create a separate WebGLContext::PresentInto function for WebXR. r=jgilbert
#   
#   - Do not apply mIsCanvasDirty bailout when presenting from a opaque FBO
#   back buffer.
#   - Create a separate WebGLContext::PresentInto with more specific XR
# ==============================================================================

diff -r 6e9a89ead3a5 -r 71ad2ed8e9ba dom/canvas/ClientWebGLContext.cpp
--- a/dom/canvas/ClientWebGLContext.cpp	Wed Jun 10 22:21:04 2020 +0000
+++ b/dom/canvas/ClientWebGLContext.cpp	Wed Jun 10 22:21:07 2020 +0000
@@ -419,7 +419,7 @@
 
 void ClientWebGLContext::Present(WebGLFramebufferJS* const fb,
                                  const layers::TextureType type) {
-  if (!mIsCanvasDirty) return;
+  if (!mIsCanvasDirty && !fb) return;
   mIsCanvasDirty = false;
   mFrontBufferSnapshot = nullptr;
 